app/grandchallenge/components/tasks.py

```diff
@@ -961,6 +963,50 @@ def add_images_to_component_interface_value(
     civ.image.update_viewer_groups_permissions()
 
 
+@shared_task
+def add_file_to_component_interface_value(
+    *,
+    component_interface_value_pk,
+    user_upload_pk,
+    target_pk,
+    target_app,
+    target_model,
+):
+    user_upload = UserUpload.objects.get(pk=user_upload_pk)
+    civ = get_model_instance(
+        pk=component_interface_value_pk,
+        app_label="components",
+        model_name="componentinterfacevalue",
+    )
+    target = get_model_instance(
+        pk=target_pk,
+        app_label=target_app,
+        model_name=target_model,
+    )
+    error = None
+    with transaction.atomic():
+        try:
+            civ.validate_user_upload(user_upload)
+            civ.full_clean()
+        except ValidationError as e:
+            transaction.set_rollback(True)
+            error = str(e)
+        else:
+            user_upload.copy_object(to_field=civ.file)
+            civ.save()
+
+    if error is not None:
+        Notification.send(
+            type=NotificationType.NotificationTypeChoices.FILE_COPY_STATUS,
+            actor=user_upload.creator,
+            message=f"File for interface {civ.interface.title} failed validation.",
+            target=target,
```

Comment:
I don't think you need a target for this type of notification, do you? The receivr is the upload session creator and th only thing that is printed is the description (i.e. the message).

Reply:
Target is not nullable: https://github.com/comic/grand-challenge.org/blob/main/app/grandchallenge/notifications/models.py#L94. But I think the notification links to the target right? So this might actually be a nice way to give some context.

Reply:
Oh right, forgot about that. For the user, the link is not visible (unless you print it explicitly), but for us admins the context is there indeed.